src/graphdb.js

```diff
@@ -52,42 +52,81 @@ function toPascalCase (str) {
     return r.trim();
 }
 
+// Changes every instance of invalid characters in the given label with the following abbreviations
+function replaceCleanseLabel(label) {
+    return label
+        .replaceAll("!", "_ex_")
+        .replaceAll("$", "_dol_")
+        .replaceAll("&", "_amp_")
+        .replaceAll("(", "_op_")
+        .replaceAll(")", "_cp_")
+        .replaceAll(".", "_dot_")
+        .replaceAll(":", "_cn_")
+        .replaceAll("=", "_eq_")
+        .replaceAll("@", "_at_")
+        .replaceAll("[", "_os_")
+        .replaceAll("]", "_cs_")
+        .replaceAll("{", "_oc_")
+        .replaceAll("|", "_vb_")
+        .replaceAll("}", "_cc_")
+        .replaceAll("-", "_hy_");
+}
 
 function graphDBInferenceSchema (graphbSchema, addMutations) {
     let r = '';
+    let invalidNode = false;
     const gdbs = JSON.parse(graphbSchema);
 
     checkForDuplicateNames(gdbs);
 
     gdbs.nodeStructures.forEach(node => {
         // node type
-        let nodeCase = node.label;
+        let nodeCase = replaceCleanseLabel(node.label);
         if (changeCase) {
-            nodeCase = toPascalCase(node.label); 
+            nodeCase = toPascalCase(nodeCase);
+        }
+
+        if (node.label !== nodeCase) {
             r += `type ${nodeCase} @alias(property:"${node.label}") {\n`;
-        } else {
+        }
+        else {
             r += `type ${nodeCase} {\n`;
         }
         
         r += '\t_id: ID! @id\n';
 
         node.properties.forEach(property => {
-            if (property.name == 'id')
+            if (property.name == 'id') {
                 r+= `\tid: ID\n`;
-            else
-                r+= `\t${property.name}: ${property.type}\n`;
+            }
+            else {
+                let propertyCase = replaceCleanseLabel(property.name);
+                let alias = '';
+                if (property.name !== propertyCase) {
+                    alias = ` @alias(property: "${property.name}")`;
+                }
+                r+= `\t${propertyCase}: ${property.type}${alias}\n`;
+            }
         });
         
         let edgeTypes = [];
         gdbs.edgeStructures.forEach(edge => {            
             edge.directions.forEach(direction => {
-                let fromCase = toPascalCase(direction.from);
-                let toCase = toPascalCase(direction.to);
-                let edgeCase = toPascalCase(edge.label);
+                nodeCase = toPascalCase(nodeCase);
```

Comment:
Does this change need to be here? Now the `nodeCase` will always be converted to pascal case regardless if `changeCase` is true or false.

Reply:
whoops you're right i'll go remove that